docs/conf.py

```diff
@@ -58,63 +49,49 @@ def __getattr__(cls, name):
-    # ecolor, capsize, and barsabove settings. So this will need to
-    # be kept up-to-date with pylab.errorbar.
-    #
-    def errorbar(self, ecolor=None, capsize=None, barsabove=False):
-        pass
-
-
 class XSPECMock(Mock):
     """Need to override get_xsversion to return a "useful" string"""
 
     def get_xsversion(self):
         "Presumably XSPEC will never reach this"
         return "999.999.999"
 
-
-# It's not clear if pylab is needed here. At present non of the
-# CIAO-related modules (e.g. the crates I/O backend) are mocked;
-# should they be?
-#
-sys.modules['astropy.io.fits'] = Mock()
-sys.modules['pylab'] = PylabMock()
-
-# This list was found by trying to import various Sherpa modules, adding
-# in mocks as necessary, and repeating. Some of these may be unnescessary.
-#
-for mod_name in ['sherpa.utils._utils',
-                 'sherpa.utils._psf',
-                 'sherpa.models._modelfcts',
-                 'sherpa.estmethods._est_funcs',
-                 'sherpa.optmethods._saoopt',
-                 'sherpa.stats._statfcts',
-                 'sherpa.astro.utils._utils',
-                 'sherpa.astro.utils._pileup',
-                 'sherpa.astro.utils._region',
-                 'sherpa.astro.utils._wcs',
-                 'sherpa.astro.models._modelfcts',
-                 'sherpa.astro.io.pyfits_backend',
+for mod_name in [
                  'sherpa.image.DS9',
-                 # Extra modules
-                 'group'
                 ]:
     sys.modules[mod_name] = Mock()
 
 # Specialized mocks
 #
 sys.modules['sherpa.astro.xspec._xspec'] = XSPECMock()
 
-# If in the docs directory then add the parent to the system path
-# so that Sherpa can be found.
-#
-if runtime_dir == 'docs':
-    sys.path.insert(0, os.path.abspath('..'))
+# If extensions (or modules to document with autodoc) are in another directory,
+# add these directories to sys.path here. If the directory is relative to the
+# documentation root, use os.path.abspath to make it absolute, like shown here.
+# sys.path.insert(0, os.path.abspath('..'))
+# IMPORTANT: the above commented section was generated by sphinx-quickstart, but
+# is *NOT* appropriate for Sherpa. It is left
+# commented out with this explanation to make it clear why this should not be
+# done. If the sys.path entry above is added,
+# it will import the *source* version of astropy instead of the
+# version installed (if invoked as "make html" or directly with sphinx), or the
+# version in the build directory.
+# Thus, any C-extensions that are needed to build the documentation will *not*
+# be accessible, and the documentation will not build correctly.
+# See astropy/docs/conf.py for details where this explanation is taken from.
+
+# -- check that Sherpa is installed ---------------------------------------
+# Since this is a change from the previous way of building the docs
+# it warrants a check and a useful error message.
+try:
+    version = metadata.version('sherpa')
```

Comment:
```suggestion
    sherpa_release = metadata.version('sherpa')
```

We use `sherpa_release` below so may as well set it here rather than change the variable name.

Reply:
Line 102 which says

```python
sherpa_release = sherpa._version.get_versions()['version']
```

can be deleted, but I can't seem to add a note to point this out (as it's unchanged code).